### PR TITLE
Update Managing Finch on Windows WSL Conf Documentation

### PR DESCRIPTION
**Pull request by subashkotha**
**Issue number:**

Closes #

**Description of changes:**
Updated WSL configuration documentation for managing Finch on Windows. Added instructions for disabling the default auto-mount behavior, editing fstab to mount the C drive in read-only mode, and steps for mounting directories in read-write mode if needed

**Terms of contribution:**

By submitting this pull request, I confirm that my contribution is made under
the terms of the licenses outlined in the LICENSE-SUMMARY file.

### CODE REVIEW

File: docs/docs/managing-finch/windows/wsl-configuration.md [hunk 1]
```diff
@@ -31,3 +31,46 @@ In addition to setting max memory limits, user's can also specify to clean up WS
 # release of cached memory.
 autoMemoryReclaim=dropcache
 ```
+**WARNING:** By default, Finch uses the standard WSL configuration, which mounts the host's C drive into Finch VM with read-write access.
+If you prefer to restrict access to the VM (and the containers running inside) by setting the C drive to read-only, follow the steps below.
```
Comment (swagatbora90) [inline docs/docs/managing-finch/windows/wsl-configuration.md:35]: Formatting
```suggestion
!!! warning
        By default, Finch uses the standard WSL configuration, which mounts the host's C drive into Finch VM with read-write access. If you prefer to restrict access to the VM (and the containers running inside) by setting the C drive to read-only, follow the steps below.
```

**Review by swagatbora90 — COMMENTED** [review]
I think the warning message should be in the section where we have Finch on Windows Installation instruction [here](https://runfinch.com/docs/managing-finch/windows/installation/#installing-finch). And have a link to a section about reconfiguring WSL config.

File: docs/docs/managing-finch/windows/wsl-configuration.md (hunk 1 above)
Comment (subashkotha) [reply docs/docs/managing-finch/windows/wsl-configuration.md:35]: Ack, will add warning message at the end of installation instructions.

**Review by pendo324 — COMMENTED** [review]
Couple of minor comments

File: docs/docs/managing-finch/windows/wsl-configuration.md [hunk 2]
```diff
@@ -31,3 +31,46 @@ In addition to setting max memory limits, user's can also specify to clean up WS
...
+To apply your changes, exit the VM and restart Finch:
+
+```
+PS C:\> wsl --shutdown 
+PS C:\> finch vm status
+Stopped
+PS C:\> finch vm start
+```
+
+Finch on Windows does not make any claims or guarantees regarding VM-level isolation, so consider the instance as the security boundary.
```
[the remark this answers is not in the document]
Comment (subashkotha) [reply docs/docs/managing-finch/windows/wsl-configuration.md:74]: Makes sense, will fix it.

File: docs/docs/managing-finch/windows/wsl-configuration.md [hunk 3]
```diff
@@ -31,3 +31,46 @@ In addition to setting max memory limits, user's can also specify to clean up WS
...
+```
+PS C:\> wsl --shutdown 
+PS C:\> finch vm status
+Stopped
+PS C:\> finch vm start
+```
+
+Finch on Windows does not make any claims or guarantees regarding VM-level isolation, so consider the instance as the security boundary. 
+The steps outlined above will help ensure that the containers operate in a more secure environment, enhancing overall security posture. 
+If you do not have such security considerations, you can continue using the default settings.
```
[the remark this answers is not in the document]
Comment (subashkotha) [reply docs/docs/managing-finch/windows/wsl-configuration.md:76]: Will remove them

**Review by pendo324 — APPROVED** [review]

**Review by swagatbora90 — APPROVED** [review]

**Pull request merged**